pages/index.tsx

```diff
@@ -45,17 +60,42 @@ const Home: NextPage = () => {
 
       <main className={styles.main}>
         <div className={styles.container}>
-          {path ? 
-            <IceGrid hexagonList={hexagonList} path={path} />
+          {result ?
+            <FrostedBox 
+              buttonLabel="Nochmal spielen"
+              description={result === "won" ? "Hurra! Du hast gewonnen!" : "Uuupsi! Du hast verloren!"}
+              onButtonClick={handleReset}
+            />
           :
-            <div className={styles.box}>
-              <div className={styles.boxChildren}>
-                <p className={styles.description}>
-                  Kannst du den Weg über&apos;s Eis wieder finden?
-                </p>
-                <button className={styles.button} onClick={() => setPath(generatePath(level))}>Start</button>
-              </div>
-            </div>
+            path ? 
+              isSolutionShown ?
```

Comment:
There's a bunch of nested ternaries here. Let's keep it in mind for some other time.